widgets/search_bar.py

```diff
@@ -102,25 +102,28 @@ def clear_all(self) -> None:
 
     def _setup_layout(self) -> None:
         self.setLayout(layout := QVBoxLayout())
+        layout.addWidget(self.bar)
         layout.addWidget(self.opts)
         layout.addWidget(self.remote_opts)
-        layout.addWidget(self.bar)
         self.setSizePolicy(QSizePolicy.Policy.MinimumExpanding, QSizePolicy.Policy.Maximum)
         self.bar.focus_search_edit()
 
-    def get_request_args(self) -> dict[str, str]:
+    def get_request_args(self, min_length: int, max_length: int) -> dict[str, str]:
         assert self._web_mode, "Web mode must be enabled."
         args = {}
         widgets = (
             self.remote_opts.sort_combo,
             self.remote_opts.category_combo,
             self.remote_opts.jlpt_level_combo,
+            self.remote_opts.wanikani_level_combo,
         )
         if keyword := self.bar.search_text():
             args["keyword"] = keyword
             for widget in widgets:
                 if param := widget.currentData().http_arg:
                     args[widget.key] = param
+            args["min_length"] = min_length
+            args["max_length"] = max_length
```

Comment:
I think this would make more sense considering that 0 stands for no limit.
```
            if min_length > 0:
                args["min_length"] = min_length
            if max_length > 0:
                args["max_length"] = max_length
```